calico_node/startup/startup.go

```diff
@@ -906,6 +918,69 @@ func ensureDefaultConfig(ctx context.Context, cfg *apiconfig.CalicoAPIConfig, c
 	return nil
 }
 
+// ensureMigrated ensures any data migration needed is done.
+func ensureMigrated(ctx context.Context, cfg *apiconfig.CalicoAPIConfig, c client.Interface) error {
+	_, err := c.ClusterInformation().Get(ctx, "default", options.GetOptions{})
+	if err == nil {
+		// Nothing to do ClusterInformation exists so data has already been migrated
+		return nil
+	} else if _, ok := err.(cerrors.ErrorResourceDoesNotExist); !ok {
+		// The error indicates a problem with accessing the resource
+		return err
+	} else {
+		_, cv1, err := clients.LoadClients("", "")
+		if err != nil {
+			return err
+		}
+		// Grab the version from the clientv1
+		v, err := getV2ClusterVersion(cv1)
+		if err != nil {
+			if _, ok := err.(cerrors.ErrorResourceDoesNotExist); ok {
+				return nil
+			} else {
+				return err
+			}
+		}
+
+		// Migrate only if it is possible to migrate from the current version
+		if canMigrate(v) {
+			_, r := migrate.Migrate(c, cv1, false) // Do migration
+			if r != ResultOk {
+				return fmt.Errors("Migration failed")
+			}
+			return nil
+		}
+	}
+	return nil
+}
+
+// Check the passed interfaces to determine the Version to decide if any
+// migration is needed.
+func getV2ClusterVersion(cv1 clients.V1ClientInterface) (string, error) {
+	if kv, err := cv1.Get(model.GlobalConfigKey{Name: "CalicoVersion"}); err == nil {
+		return kv.Value.(string), nil
+	} else {
+		return "", err
+	}
+}
+
+// Checks the version string passed in if it possible to migrate from that
```

Comment:
typo: "in if it possible"

Reply:
nit: Golang function comments should start with the function name.

e.g.

"canMigrate returns true if the given version. . ."